sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java

```diff
@@ -348,44 +367,38 @@ private HttpPipeline createPipeline() {
 
         // Closest to API goes first, closest to wire goes last.
         final List<HttpPipelinePolicy> httpPolicies = new ArrayList<>();
-        final String clientName = properties.getOrDefault("name", "UnknownName");
-        final String clientVersion = properties.getOrDefault("version", "UnknownVersion");
 
         // Find applicationId to use
-        String logApplicationId = null;
-        if (httpLogOptions != null) {
-            logApplicationId = httpLogOptions.getApplicationId();
-        }
+        final String applicationId = CoreUtils.getApplicationId(clientOptions, httpLogOptions);
 
-        String clientApplicationId = null;
-        if (clientOptions != null && clientOptions.getApplicationId() != null) {
-            clientApplicationId = clientOptions.getApplicationId();
-        }
-
-        if (logApplicationId != null && clientApplicationId != null
-            && !logApplicationId.equalsIgnoreCase(clientApplicationId)) {
-            throw logger.logExceptionAsError(new IllegalStateException(
-                "'httpLogOptions.getApplicationId() and clientOptions.getApplicationId()' cannot be different."));
-        }
-        // We prioritize application id set in ClientOptions.
-        final String applicationId = clientApplicationId != null ? clientApplicationId : logApplicationId;
-
-        httpPolicies.add(new UserAgentPolicy(applicationId, clientName, clientVersion,
-            buildConfiguration));
+        httpPolicies.add(new UserAgentPolicy(applicationId, CLIENT_NAME, CLIENT_VERSION, buildConfiguration));
         httpPolicies.add(new ServiceBusTokenCredentialHttpPolicy(tokenCredential));
         httpPolicies.add(new AddHeadersFromContextPolicy());
 
+        httpPolicies.addAll(perCallPolicies);
+
         HttpPolicyProviders.addBeforeRetryPolicies(httpPolicies);
 
         httpPolicies.add(retryPolicy == null ? new RetryPolicy() : retryPolicy);
-        httpPolicies.addAll(userPolicies);
+        httpPolicies.addAll(perRetryPolicies);
+
+        if (clientOptions != null) {
+            List<HttpHeader> httpHeaderList = new ArrayList<>();
+            clientOptions.getHeaders().forEach(h -> httpHeaderList.add(new HttpHeader(h.getName(), h.getValue())));
+
+            if (!httpHeaderList.isEmpty()) {
+                httpPolicies.add(new AddHeadersPolicy(new HttpHeaders(httpHeaderList)));
+            }
+        }
```

Comment:
should this be using the CoreUtils method added above?

Reply:
I'll let libraries update once azure-core ships the API rather than using unreleased dependencies.